### ISSUE TITLE
fatal error: concurrent map writes

### ISSUE THREAD

**Issue by romanlevin** [moby/moby#35649]
**Description**

When running Docker on a Vagrant box based off of ubuntu/xenial64, all containers hang and Docker can't stop them. Looking at the daemon logs reveals a fatal error (attached below). Though the logs state that the Daemon has been restarted, the problem persists until `service docker restart`.

My host machine is a mid-2015 MacBook Pro 15" running High Sierra.

**Steps to reproduce the issue:**
Seems to happen randomly, but at least twice a day. As far as I can tell this only happens while running a Docker Compose swarm, but the two might not be related because I usually run containers using Docker Compose.

**Output of `docker version`:**

```
Client:
 Version:      17.11.0-ce
 API version:  1.34
 Go version:   go1.8.3
 Git commit:   1caf76c
 Built:        Mon Nov 20 18:37:39 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.11.0-ce
 API version:  1.34 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   1caf76c
 Built:        Mon Nov 20 18:36:09 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 15
 Running: 15
 Paused: 0
 Stopped: 0
Images: 140
Server Version: 17.11.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 992280e8e265f491f7a624ab82f3e238be086e49
runc version: 0351df1c5a66838d0c392b4ac4cf9450de844e2d
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-101-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 3.859GiB
Name: turbo-dev
ID: 6JA3:MIAP:ICRO:OGAD:HRKZ:LIN5:GPUM:LA3T:AZD3:7OZF:EBYO:7VZ2
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Docker log excerpt**
```
Nov 30 11:00:03 turbo-dev dockerd[1197]: fatal error: concurrent map writes
Nov 30 11:00:03 turbo-dev dockerd[1197]: goroutine 21510 [running]:
Nov 30 11:00:03 turbo-dev dockerd[1197]: runtime.throw(0x2a8410d, 0x15)
Nov 30 11:00:03 turbo-dev dockerd[1197]:         /usr/local/go/src/runtime/panic.go:596 +0x95 fp=0xc420a9d7d8 sp=0xc420a9d7b8
Nov 30 11:00:03 turbo-dev dockerd[1197]: runtime.mapassign(0x27991a0, 0xc4201f2ea0, 0xc420a9d8b8, 0x40)
Nov 30 11:00:03 turbo-dev dockerd[1197]:         /usr/local/go/src/runtime/hashmap.go:499 +0x667 fp=0xc420a9d878 sp=0xc420a9d7d8
Nov 30 11:00:03 turbo-dev dockerd[1197]: github.com/docker/docker/daemon.(*Daemon).registerExecPidUnlocked(0xc42046c800, 0xc4212146c0, 0xc4200d5200)
Nov 30 11:00:03 turbo-dev dockerd[1197]:         /go/src/github.com/docker/docker/daemon/exec.go:39 +0x218 fp=0xc420a9d918 sp=0xc420a9d878
...
```

The full stack trace is here: https://pastebin.com/7DxBSs0v

**Comment by tha80** [comment]
I also have the exact same problem on my Debian Jessie Box with Docker 17.11.0-ce (without swarm) and on another box with Debian Stretch an 17.11.0-ce in Swarm Mode.

All I can do to fix this is to restart the whole system. Docker 17.11. doesn't seem to be that stable. Please let me know if specific information is needed to help debug this problem.

**Comment by thaJeztah** [comment]
This is a duplicate of https://github.com/moby/moby/issues/35588, which is fixed on master through https://github.com/moby/moby/pull/35614 and will be included in Docker 17.12 (release candidate 1 for that is currently available in the "test" channel)


Closing this issue because this is resolved in this repository, but thanks for reporting, and feel free to comment after I closed 👍
